### PR TITLE
MULE-18073: Non-blocking operations don't switch out of the selector

### PR DESCRIPTION
**Pull request by elrodro83**
thread after execution

### CODE REVIEW

File: modules/extensions-spring-support/src/test/resources/iron-man-config.xml [hunk 1]
```diff
@@ -22,16 +22,30 @@
     <flow name="fireMissile">
         <test:assert-intercepting responseSameTask="false" />
         <marvel:fire-missile at="#[payload]" config-ref="ironMan"/>
+        
+        <custom-processor class="org.mule.test.module.extension.nb.NonBlockingOperationsTestCase$ThreadCaptor" />
+        <error-handler>
+            <on-error-propagate>
+                <logger/>
+                <!-- MULE-18118 -->
+                <!-- <custom-processor class="org.mule.test.module.extension.nb.NonBlockingOperationsTestCase$ThreadCaptor" /> -->
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/resources/iron-man-config.xml]: remove or uncomment

File: modules/extensions-spring-support/src/test/resources/iron-man-config.xml [hunk 2]
```diff
@@ -22,16 +22,30 @@
...
+
+        <custom-processor class="org.mule.test.module.extension.nb.NonBlockingOperationsTestCase$ThreadCaptor" />
     </flow>
 
     <flow name="computeFlightPlan">
         <test:assert-intercepting responseSameTask="false" />
         <marvel:compute-flight-plan config-ref="ironMan" />
+
+        <!-- MULE-18119 -->
+        <!-- <custom-processor class="org.mule.test.module.extension.nb.NonBlockingOperationsTestCase$ThreadCaptor" /> -->
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/resources/iron-man-config.xml]: same

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Pull request merged**